hibernate-testing/src/main/java/org/hibernate/testing/orm/junit/EntityManagerFactoryExtension.java

```diff
@@ -176,16 +175,14 @@ public static EntityManagerFactoryScope findEntityManagerFactoryScope(
 			integrationSettings.put( setting.name(), setting.value() );
 		}
 
-		if ( emfAnn.nonStringValueSettingProviders().length > 0 ) {
-			for ( int i = 0; i < emfAnn.nonStringValueSettingProviders().length; i++ ) {
-				final Class<? extends NonStringValueSettingProvider> _class = emfAnn.nonStringValueSettingProviders()[i];
-				try {
-					NonStringValueSettingProvider valueProvider = _class.newInstance();
-					integrationSettings.put( valueProvider.getKey(), valueProvider.getValue() );
-				}
-				catch (Exception e) {
-					log.error( "Error obtaining special value for " + _class.getName(), e );
-				}
+		for ( SettingProvider providerAnn : emfAnn.settingProviders() ) {
+			final Class<? extends SettingProvider.Provider<?>> providerImpl = providerAnn.provider();
```

Comment:
I think this should be `final Class<? extends SettingProvider.Provider> providerImpl = providerAnn.provider();`